bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -176,6 +176,10 @@ void ReplicaImp::onMessage<ClientRequestMsg>(ClientRequestMsg *m) {
   const bool invalidClient = !isValidClient(clientId);
   const bool sentFromReplicaToNonPrimary = repsInfo->isIdOfReplica(senderId) && !isCurrentPrimary();
 
+  if (invalidClient) {
+    clientsManager->getInvalidClients().insert(clientId);
```

Comment:
I would just add to a counter member variable as done for noops. No need to maintain a set. The goal is to count how frequently invalid clients arrive to track possible bugs or attacks, not to know what the IDs are. We could always log them if we wanted to.

Reply:
Yes, changed it to numerical counter. If we need actual IDs, we can log them later. Thanks